### ISSUE TITLE
MissingFieldException and MissingMethodException when using Packer

### ISSUE THREAD

**Issue by Drizin** [yck1509/ConfuserEx#478]
I've just started using ConfuserEx, and found something that I believe is a bug when **Packing assemblies with Inner Classes**:

Defined global obfuscation in AssemblyInfo:
```csharp
[assembly: Obfuscation(Exclude = false, Feature = "anti debug;anti ildasm;constants(elements=SNPI);rename(mode=sequential)")]
[assembly: Obfuscation(Exclude = false, Feature = "preset(minimum);+ctrl flow;-anti debug;+rename(mode=letters,flatten=false);")]
```

Defined some classes to not be renamed because I use them in serialization:
```csharp
    [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
    public class SerializedClass
    {
        public SerializedClass()
        {
        }

        public ExternalClass ExternalObject; // external class works fine

        // problem 1 - MissingFieldException if pointing to an inner class
        public InnerClass InnerObject; 
        [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
        public class InnerClass
        {
        }

        // problem2 - MissingFieldException if pointing to an inner enum
        [System.Xml.Serialization.XmlAttribute] public InnerEnum state = InnerEnum.State0;
        [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
        public enum InnerEnum
        {
            State0,
            State1,
        }

        // problem3 - MissingMethodException if getter is using internal class
        [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
        public ListOfStrings MyStrings
        {
            get { return new ListOfStrings(); }
        }
        [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
        public class ListOfStrings : List<string>
        {

        }

    }

    [ObfuscationAttribute(Exclude = false, Feature = "-rename")]
    public class ExternalClass
    {
    }
```

Main block:
```csharp
        static void Main(string[] args)
        {
            var obj = new SerializedClass();
            XmlSerializer xsSubmit = new XmlSerializer(typeof(SerializedClass));
            using (StringWriter sww = new StringWriter())
            using (XmlWriter writer = XmlWriter.Create(sww))
            {
                xsSubmit.Serialize(writer, obj);
                Console.WriteLine(sww.ToString());
            }
            Console.ReadLine();
        }

```

As you may notice from the code above, **every time that I use an inner class (or inner enum) the respective Field pointing to the inner class/enum is not generated (or preserved) giving a MissingFieldException, and the respective getter gives a MissingMethodException**.

If someone can shed some light about where to fix, I can help fix the problem.

PS: I believe that when defining `-rename` for a class, all members automatically inherit that setting. However even if applying `-rename` directly to `InnerObject` still doesn't solve the problem.

**Comment by Drizin** [comment]
I did some more tests with the sample above, and realized that **when using `compat=true` the serialization problem doesn't happen**:

```csharp
[assembly: Obfuscation(Exclude = false, Feature = "packer:compressor(compat=true)")]
```

However, that `compat=true` doesn't work with my WPF application, giving me `System.IO.IOException: Cannot locate resource 'mainwindow.xaml' at MS.Internal.AppModel.ResourcePath.GetStreamCore(FileMode mode...`.

I also tried `preset(none)`, and even `+rename(renXaml=false)`, but to no avail.

In summary: **If I use packer, serialization of inner classes doesn't work. If I use packer with `compat=true`, WPF doesn't launch**.

My **workaround** was removing inner classes and inner enums out of the serialized classes.
